### ISSUE TITLE
Incompatible tensor shape in FFA.py?

### ISSUE THREAD

**Issue by charles92** [zhilin007/FFA-Net#18]
Hi,

Thanks for open-sourcing your work. I was trying to write a TensorFlow implementation of your model, and ran into the following issue.

In `FFA.py`, you have the code below at [L96](https://github.com/zhilin007/FFA-Net/blob/34ab67a2a10203462ce105c19aa20d5214777038/net/models/FFA.py#L96):

```py
w=w.view(-1,self.gps,self.dim)[:,:,:,None,None]
out=w[:,0,::]*res1+w[:,1,::]*res2+w[:,2,::]*res3
```

### Questions:

1. After the first line, tensor `w` will have shape `[B, gps, dim, 1, 1]`, where `B` is the batch size. Is that correct?
2. Subsequently, in the second line, `w[:, 0, ::]` has shape `[B, dim, 1, 1]`. Correct?
3. But `res1` has shape `[B, H, W, dim]`, where `H` and `W` are the spatial dimensions. How could you multiply `w[:, 0, ::]` and `res1` which have incompatible shapes?

Sorry if this is trivial - I normally write TensorFlow and am quite unfamiliar with the specifics about PyTorch. Thanks!

**Comment by charles92** [comment]
I asked a few Torch experts and they pointed out that Torch uses `[B, C, H, W]` convention for their `Conv2d` outputs. Hence, `res1` has shape `[B, dim, H, W]`, which matches the dimensions of `w`. There's no issue with the Torch code.
